packages/mock-app/src/pages/Verify.tsx

```diff
@@ -54,15 +54,22 @@ const Verify = () => {
       const { payload } = JSON.parse(payloadQuery);
       const { uri, key, hash } = payload;
       const encryptedCredential = await publicAPI.get(uri);
-      if (encryptedCredential?.credentialSubject) {
-        return setCredential(encryptedCredential);
-      }
+
+      const verifyHash = (credential: VerifiableCredential) => {
+        if (hash) {
+          const computedHash = computeHash(credential);
+          if (computedHash !== hash) return displayErrorUI(['Hash invalid']);
+        }
+
+        return setCredential(credential);
+      };
 
       if (
-        encryptedCredential?.type?.includes('EnvelopedVerifiableCredential') &&
-        encryptedCredential?.id?.startsWith('data:application/')
+        encryptedCredential?.credentialSubject ||
+        (encryptedCredential?.type?.includes('EnvelopedVerifiableCredential') &&
+          encryptedCredential?.id?.startsWith('data:application/'))
       ) {
-        return setCredential(encryptedCredential);
+        return verifyHash(encryptedCredential);
```

Comment:
@ldhyen99 the code looks like we are not handling decrypting the enveloped credential. Should we add the decryption for it?